packages/nc-gui/components/cell/GeoData.vue

```diff
@@ -0,0 +1,154 @@
+  formState.longitude = longitude
+}
+
+const onClickSetCurrentLocation = () => {
+  isLoading = true
+  const onSuccess = (position) => {
+    const crd = position.coords
+    console.log('crd', crd)
+    formState.latitude = crd.latitude
+    formState.longitude = crd.longitude
+    isLoading = false
+  }
+
+  const onError = (err) => {
+    console.warn(`ERROR(${err.code}): ${err.message}`)
+    isLoading = false
+  }
+
+  const options = {
+    enableHighAccuracy: true,
+    timeout: 20000,
+    maximumAge: 2000,
+  }
+  navigator.geolocation.getCurrentPosition(onSuccess, onError, options)
+}
+</script>
+
+<template>
+  <a-dropdown :is="isExpanded ? AModal : 'div'" v-model:visible="isExpanded" trigger="click">
+    <a-button>{{ latLongStr }}</a-button>
+    <template #overlay>
+      <a-form :model="formState" class="flex flex-col" @finish="handleFinish">
+        <a-form-item class="inputLat">
+          <template #label>
+            <div class="fixed-width">{{ $t('labels.lat') }}</div>
+          </template>
+          <a-input
+            v-model:value="formState.latitude"
+            type="number"
+            step="0.0000001"
+            :min="-90"
+            required
+            :max="90"
+            @keydown.stop
+            @selectstart.capture.stop
+            @mousedown.stop
+          />
+        </a-form-item>
+
+        <a-form-item class="inputLng">
+          <template #label>
+            <div>{{ $t('labels.lng') }}</div>
+          </template>
+          <a-input
+            v-model:value="formState.longitude"
+            type="number"
+            step="0.0000001"
+            required
+            :min="-180"
+            :max="180"
+            @keydown.stop
+            @selectstart.capture.stop
+            @mousedown.stop
+          />
+        </a-form-item>
+        <a-form-item>
+          <div style="display: flex; align-items: center; margin-right: 0.5rem">
+            <MdiReload v-if="isLoading" :class="{ 'animate-infinite animate-spin text-gray-500': isLoading }" />
+            <a-button class="ml-2" @click="onClickSetCurrentLocation">{{ $t('labels.yourLocation') }}</a-button>
+          </div>
+        </a-form-item>
+        <a-form-item class="buttons">
+          <a-button type="text" @click="clear">{{ $t('general.cancel') }}</a-button>
+          <a-button type="primary" html-type="submit">{{ $t('general.submit') }}</a-button>
+        </a-form-item>
+      </a-form>
+    </template>
+  </a-dropdown>
+</template>
+
+<style scoped lang="scss">
+input[type='number']:focus {
+  @apply ring-transparent;
+}
+
+.inputLat {
+  width: 180px;
+  margin-top: 1rem;
+  margin-bottom: 0.5rem;
+  margin-right: 0.5rem;
+  margin-left: 0.5rem;
+}
+.inputLng {
+  width: 180px;
+  margin-right: 0.5rem;
+  margin-left: 0.5rem;
+}
+
+.fixed-width {
+  width: 24px;
+}
+.buttons {
+  margin-left: auto;
+  margin-bottom: 0;
+}
+.ant-dropdown-menu {
+  height: fit-content;
+  align-items: flex-end;
+}
```

Comment:
prefer windicss style like `input[type='number']:focus` one

Reply:
Here I'm not sure what you exactly mean can you elaborate?

Reply:
My understanding is that you prefer a very generic, element type focused selector approach, but I think need classes to add element specific adjustments. Can you give a concrete example where you would rewrite a selector?